Allow showing empty tanks via Fluid Bars

MalTeeez opened the pull request:
This PR allows the Fluid Bars introduced with #43 to also render empty tanks.

Other part of functionality provided in https://github.com/GTNewHorizons/WAILAPlugins/issues/23, but there is no hard dependency for it.
Since #49 added some more changes to the renderer, I based this PR on that, which is therefore required before merging.

I also added some runtimeOnly dependencies to make it easier to test out this and other features.

MalTeeez committed:
Accept empty fluid tanks and treat them differently in fluid bars

(cherry picked from commit 7cb54e77707b844aae2f379e1759a59640dd7221)

MalTeeez committed:
Only render fluid when not empty, change colors a little to fit empty bar better

MalTeeez commented on src/main/java/mcp/mobius/waila/overlay/tooltiprenderers/TTRenderFluidBar.java:
```diff
@@ -63,68 +65,98 @@ public Dimension getSize(String[] params, IWailaCommonAccessor accessor) {
-                OverlayConfig.fontcolor,
-                true);
 
+        drawThickBeveledBox(0, 0, maxStringW, height, 1, 0xFF787878, 0xFF787878, -1);
```
No here it doesn't make as much of an impact as it does for the alpha on the gradient - I'll switch that back

SuperSoupr commented on src/main/java/mcp/mobius/waila/overlay/tooltiprenderers/TTRenderFluidBar.java:
```diff
@@ -63,68 +65,98 @@ public Dimension getSize(String[] params, IWailaCommonAccessor accessor) {
+                    true);
+        } else {
+            DisplayUtil.drawString(
+                    LangUtil.translateG("hud.msg.empty").replace("<", "").replace(">", "") + " / "
```
Just add another lang string, no need to do the .replaces

MalTeeez committed:
Change box color back, consider empty tanks in getSize() too

(cherry picked from commit cc9861c44bf0cd5e60e646a4522411525fd98ca2)

MalTeeez committed:
Add another 'Empty' localization entry without the <> and use that instead

(cherry picked from commit f0b35d061b0dd1df6e42f48e322789f3a615318e)

MalTeeez committed:
Fix italian language change, use correct double for size calculation, switch back background opacity

Use old version of background opacity for filled tanks since we now treat bot of those as separate backgrounds

MalTeeez committed:
Switch to String.format for tooltip strings, remove wrong padding for getsize on empty tanks

MalTeeez committed:
Revert language changes to be compatible with #54

danyadev commented on src/main/java/mcp/mobius/waila/overlay/tooltiprenderers/TTRenderFluidBar.java:
```diff
@@ -49,22 +56,39 @@ public TTRenderFluidBar() {
+        String barText;
+        if (!(params[0].equals("EMPTYFLUID") && params[1].equals("EMPTYFLUID"))) {
+            int commaCount = (int) (Math.floor((params[2].length() - 1) / 3D)
+                    + Math.floor((params[3].length() - 1) / 3D));
```
wait, can't we just use `formatNumber(params[2])` & `formatNumber(params[3])` which will place commas automatically instead of calculating their amount?

btw NumberFormat doc says that grouping size is different depending on locale so current solution could lead to problems

Dream-Master committed:
Merge branch 'master' into master

MalTeeez committed:
Merge branch 'master' of github.com:GTNewHorizons/waila

MalTeeez committed:
Update to use #54 methods & address review

MalTeeez committed:
Apply spotless

MalTeeez committed:
Use same deps as #55

RaidhoSketch commented:
<img width="1920" height="1017" alt="2025-12-04_01 15 25" src="https://github.com/user-attachments/assets/bd2c85c1-1f59-448b-bf6b-4edd4dd81ac5" />
Happens with RotaryCraft's engines, despite them possessing only a single fluid tank under normal circumstances. Which is probably due to Reika's implementation, but is unwelcome behaviour in any case. Make it a config, maybe?

MalTeeez commented:
> <img alt="2025-12-04_01 15 25" width="1920" height="1017" src="https://private-user-images.githubusercontent.com/73392166/522135003-bd2c85c1-1f59-448b-bf6b-4edd4dd81ac5.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ4MDkyODgsIm5iZiI6MTc2NDgwODk4OCwicGF0aCI6Ii83MzM5MjE2Ni81MjIxMzUwMDMtYmQyYzg1YzEtMWY1OS00NDhiLWJmNmItNGVkZDRkZDgxYWM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEyMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMjA0VDAwNDMwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY5MDVkYzE5NWY1OWNjZjc5ZGJkYWNiYmE0ZWQxZWFhMmJmY2Q2NjA0MjkwZDdmNTE4MTQxYjE3N2M4MDc3YjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.eJp_SA7Okblahg17v_0z-HjNpnxOv6JSoDUY8jFBhvY"> Happens with RotaryCraft's engines, despite them possessing only a single fluid tank under normal circumstances. Which is probably due to Reika's implementation, but is unwelcome behaviour in any case. Make it a config, maybe?

That is because Reika outfits all her engines with 3 fluid tanks and then filters her own waila body by the engine type later on: https://github.com/ReikaKalseki/RotaryCraft/blob/b2288638f7f4703bb4555f72a2b86f72407cc2ae/Base/BlockBasicMachine.java#L338

MalTeeez commented:
Nonetheless a config option might be a good idea, since this functionality will probably reveal more weird TE fluid tank implementations (with unused tanks). 
It would have to be in WailaPlugins tho

RaidhoSketch commented:
Also, your chromatifixes config option does not disable the "Fuel:" parts of engines in Waila display! Which is also tank info.

MalTeeez commented:
> Also, your chromatifixes config option does not disable the "Fuel:" parts of engines in Waila display! Which is also tank info.

Yep that is mostly intentional, since Reika provides naming to those tanks (Based on the type of the TE) that we can't infer from a generic TankInfo[], so I only disabled it for the tanks where we basically provide the same info via the generic Waila handler.
